### ISSUE TITLE
Enhancements to https://github.com/gluster/gluster-csi-driver/pull/11

### ISSUE THREAD

**Issue by humblec** [gluster/gluster-csi-driver#12]
The below tasks are missing in https://github.com/gluster/gluster-csi-driver/pull/11

- [x] checkvolume list need to be enabled
- [x] volume has to be deleted when voulme start failed
- [x] volume  has to be deleted when fetching of cluster nodes failed
- [x] delete volume need to return success if the volume is not present.

**Comment by JohnStrunk** [comment]
What is the rationale for having these items as future work instead of fixing them directly in #11?

I can understand opening issues for enhancements that were suggested as a part of the review, but some of these look like straight-up bugs (e.g., delete when start fails). A "happy-path only" code drop is not going to get us high quality.

**Comment by humblec** [comment]
@JohnStrunk the issue was planned for a zero day PR with #11 , The issues were opened and assigned to have this worked in parallel. It does not mean we are dropping high quality when we `productize it`.

**Comment by Madhu-1** [comment]
>The below tasks are missing in #11
   checkvolume list need to be enabled
  volume has to be deleted when voulme start failed
  volume has to be deleted when fetching of cluster nodes failed
  delete volume need to return success if the volume is not present.

@humblec i have addressed the first and last issue already in the PR,
i think we dont need to delete the volume if we fail to start the volume or if cluster nodes are not present
as we are enabling the checkvolume functionality, we are picking the same volume again and starting it again if its not started

**Comment by humblec** [comment]
Agree.. thats enough.. thanks !

**Comment by humblec** [comment]
Fixed @ https://github.com/gluster/gluster-csi-driver/pull/16

**Comment by Madhu-1** [comment]
@humblec i think this one also can be closed

**Comment by humblec** [comment]
yes, closing it!

**Comment by Madhu-1** [comment]
@humblec this needs to be closed

**Comment by humblec** [comment]
This is done, thanks!
